snekmer/scripts/learn_evaluate_sequences.py

```diff
@@ -0,0 +1,709 @@
+            )
+
+        updatedData.fillna(0, inplace=True)
+        return updatedData
+
+    def saveResults(
+        self,
+        ratioTotalDist,
+        outputPath,
+    ):
+        """
+Saves the computed results to specified paths.
+
+Args:
+    ratioTotalDist (Series): Ratio distribution to be saved.
+    outputPath (str): Path to save the ratio distribution.
+
+"""
+        ratioTotalDist.to_csv(outputPath)
+        return
+
+    def executeAll(self):
+        """
+Executes all functions in order.
+
+This method does the following:
+
+1. Processes each input file by reading, transforming, and storing intermediate variables.
+2. After processing all files, generates a global crosstab to summarize the data.
+3. Calculates distributions for True and False predictions. If base confidence files are provided,
+it merges the calculated distributions with the base ones.
+4. Computes a ratio distribution based on the True and False total distributions.
+5. Finally, writes the ratio distribution and global crosstab to csv.
+"""
+        self.generateInputs()
+        ratioCrosstab = self.generateGlobalCrosstab()
+        trueDist, falseDist = self.calculateDistributions()
+        ratioDist, total_sum, inter_sum = self.computeRatioDistribution(
+            trueDist, falseDist
+        )
+        globalConfidence = self.checkConfidenceMerge(
+            ratioDist, total_sum, inter_sum
+        )
+        self.saveResults(globalConfidence, self.outputGlob)
+
+    def generateDiffDataframe(
+        self, two_key_vals, predictions, deltas, result, tf, known
+    ):
+        """
+Generates a dataframe showing the differences and other metrics.
+
+Args:
+    two_key_vals (array-like): Two Key stored values, this can be top two scores, or top score and threshold, etc
+    predictions (Series): Predictions made by the selection method.
+    deltas (Series): Delta values calculated by the selection method.
+    result (list): Actual labels.
+    tf (list): List of "True/False" labels.
+    known (list): List of "Known/Unknown" labels.
+
+Returns:
+    DataFrame: Constructed dataframe with computed differences and other metrics.
+"""
+        # rounded_deltas = [ round(num, 2) for num in deltas ]
+        rounded_deltas = [
+            round(num, 2) if num is not None else 0 for num in deltas
+        ]
+
+        # print(f"two_key_vals in generateDiffDataframe:\n {two_key_vals}")
+        diffDataframe = pd.DataFrame(
+            {
+                "Top": two_key_vals["keyValueOne"],
+                "Second": two_key_vals["keyValueTwo"],
+                "Difference": rounded_deltas,
+                "Prediction": predictions,
+                "Actual": result,
+                "T/F": tf,
+                "Known/Unknown": known,
+            }
+        )
+
+        tmp = pd.to_numeric(diffDataframe["Difference"], errors="coerce")
+        diffDataframe["Difference"] = tmp
+
+        return diffDataframe
+
+    def apply_selectionMethod(
```

Comment:
This function is very long (~250 lines) with multiple if/elif conditionals and not much commentary on what each stream is doing. This needs to be broken down into multiple functions and better documented to describe steps